rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1760,8 +1769,33 @@ public Response listStudentContactLogEntriesByStudent(@PathParam("STUDENTID") Lo
     Student student = studentController.findStudentById(studentId);
     Status studentStatus = checkStudent(student);
     if (studentStatus != Status.OK)
-      return Response.status(studentStatus).build();
-
+      return Response.status(studentStatus).build(); 
+    
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    Boolean amIGuidanceCounselor = studentController.amIGuidanceCouncelor(studentId, staffMember);
+
+    if (!loggedUser.getRole().equals(Role.ADMINISTRATOR)) {
+      if (!amIGuidanceCounselor) {
+      // If logged user is teacher, return only their own entries
+        if (loggedUser.getRole().equals(Role.TEACHER)) {
+          List<StudentContactLogEntry> ownEntries = new ArrayList<>();
+          List<StudentContactLogEntry> entries = studentContactLogEntryController.listContactLogEntriesByStudent(student);
+          
+          for (StudentContactLogEntry entry : entries) {
+            if (entry.getCreatorId() != null) {
+              if (entry.getCreatorId().equals(loggedUser.getId())){
+                ownEntries.add(entry);
```

Comment:
The REST model of `StudentContactLogEntry` could just as well have a list of its comments so that if a client asks for contact log entries of some student and gets, say, ten of them, there wouldn't be need for ten additional calls to get the comments of each.

Here we are already figuring out which entries are returned, so this would be a good place to go through the comments as well and figure out which of them are returned.